### ISSUE TITLE
InoReader: Synchronized creation dates wrong

### ISSUE THREAD

**Issue by Skycoder42** [martinrotter/rssguard#186]
#### Brief description of the issue.

Dates in the feed list are completly broken. Wrong day, year and time for all entries:
![image](https://user-images.githubusercontent.com/9221346/39571178-f16c7812-4eca-11e8-9fa9-695c0543c236.png)

#### How to reproduce the bug?

1. Add a InoReader account to RSS Guard
2. Synchronize the account
3. Feeds in the list now show wrong dates

#### What is the expected result?

The creation date should be the actual, real date of the feed, as found in the feed rss (or in this case: in the InoReader app)

#### What actually happened?

The date is wrong. It seems to be by an offset. Consider the following 2 example dates - The distance between the two dates is the same (exactly 1046 minutes):

Actual Feed Date | RSS Guard Date
------------------------|------------------------
2018-05-02 17:49:00 | 1970-01-07 13:19:09
2018-05-03 11:15:00 | 1970-01-08 06:45:09

However, all dates in all my feeds are either in december 1969 or janurary 1970 - so this offset seems to be inconsistent.

#### Other information (logs, see Wiki)

- I am using the official build from Arch-Linux (See https://git.archlinux.org/svntogit/community.git/tree/trunk/PKGBUILD?h=packages/rssguard) - it seems to be a normal build with no special configurations
- I am synchronizing an InoReader account with Google login
- I generated a log by starting, performing a sync and quitting. There were 5 new entries for that sync run. Log: [log.txt](https://github.com/martinrotter/rssguard/files/1970478/log.txt)
- I checked the dates in the InoReader webapp itself, they are correct there
- It happens on both of my machines. Later I will check if the dates are consistently wrong or random per machine and update this issue accordingly.

**Comment by Skycoder42** [comment]
Anything?

**Comment by martinrotter** [comment]
Hello Skycoder.

Honestly, RSS Guard is no longer my priority because I changed my workflow and I do not use any RSS reader app anymore. I might fix this in a future, but I cannot guarantee anything. My time is highly limited and RSS Guard would seriously need some code contrubitors and/or maintainers.

**Comment by martinrotter** [comment]
Yes, that is exactly the place. The input value is read from downloaded JSON file, then the code must convert the value into `QDateTime` instance, ideally to UTC.

**Comment by Skycoder42** [comment]
The most classy errors of all - a buffer overflow it is!

multiplying the timestamp with 1000 overflows the 4 byte integer, so casting to qint64 beforehand would solve the problem.

Instead I changed the method to directly use the seconds to avoid this problem all together. Synchronizing the database once repairs all the stamps. I will propose a pullrequest for this now
